scos-10-build-test-qemu CI failing with:

```
=== RUN   iso.iso-offline-install-iscsi.manual.bios
=== RUN   iso.iso-fips.uefi
=== RUN   iso.iso-offline-install-iscsi.ibft.uefi
=== RUN   iso.iso-offline-install-iscsi.ibft-with-mpath.bios
Detected development build; disabling signature verification
--- FAIL: iso.iso-offline-install-iscsi.manual.bios (227.19s)
        live-iscsi.go:199: unexpected string from completion channel: "coreos-installer-test-entered-emergency-target", expected: "iscsi-boot-ok"
Detected development build; disabling signature verification
--- FAIL: iso.iso-offline-install-iscsi.ibft-with-mpath.bios (228.77s)
        live-iscsi.go:199: unexpected string from completion channel: "coreos-installer-test-entered-emergency-target", expected: "iscsi-boot-ok"
Detected development build; disabling signature verification
Warning: Cannot announce submounts, client does not support it
--- FAIL: iso.iso-offline-install-iscsi.ibft.uefi (183.52s)
        live-iscsi.go:199: unexpected string from completion channel: "coreos-installer-test-entered-emergency-target", expected: "iscsi-boot-ok"
Detected development build; disabling signature verification
--- FAIL: iso.iso-fips.uefi (604.64s)
        live-fips.go:103: Unable to create test machine: machine "f65466c7-7b65-4cb1-a09e-aa2ec19af6b0" failed to start: ssh journalctl failed: time limit exceeded
FAIL, output in /logs/artifacts/kola-testiso/rerun
```

--- a/ci/prow-entrypoint.sh
+++ b/ci/prow-entrypoint.sh
@@ -74,31 +74,23 @@ prepare_repos() {
     src/config/ci/get-ocp-repo.sh src/config/ocp.repo "${variant}"
 }
 
-# Do a cosa build only.
+# Build OS container and bootimage artifacts
 # This is called both as part of the build phase and test phase in Prow thus we
 # can not do any kola testing in this function.
-# We do not build the QEMU image here as we don't need it in the pure container
-# test case.
 cosa_build() {
     cosa fetch # used for the non build-with-buildah path
     cosa build
+    cosa osbuild qemu metal metal4k live
+    cosa compress --artifact=metal --artifact=metal4k
 }
 
-# Build QEMU image and run all kola tests
+# Run all kola tests
 kola_test_qemu() {
-    cosa osbuild qemu
     cosa kola run --parallel 2 --output-dir ${ARTIFACT_DIR:-/tmp}/kola --rerun --allow-rerun-success tags=needs-internet
 }
 
-# Build metal, metal4k & live images and run kola tests
+# Run kola iso.* tests
 kola_test_metal() {
-    # Build metal + installer now so we can test them
-    cosa osbuild metal metal4k live
-
-    # Compress the metal and metal4k images now so we're testing
-    # installs with the image format we ship
-    cosa compress --artifact=metal --artifact=metal4k
-
     # Use 'testiso' for older cosa builds, otherwise use kola.
     # Until RHCOS openshift cluster gets updated, iso.* tests run sequentially.
     if cosa kola help | grep -q testiso; then

--- a/ci/prow-entrypoint.sh
+++ b/ci/prow-entrypoint.sh
@@ -99,8 +99,15 @@ kola_test_metal() {
     # installs with the image format we ship
     cosa compress --artifact=metal --artifact=metal4k
 
-    # Run all testiso scenarios on metal artifact
-    kola testiso -S --output-dir ${ARTIFACT_DIR:-/tmp}/kola-testiso  --denylist-test iso-offline-install-iscsi* --denylist-test pxe-offline-install.rootfs-appended.bios
+    # Use 'testiso' for older cosa builds, otherwise use kola.
+    # Until RHCOS openshift cluster gets updated, iso.* tests run sequentially.
+    if cosa kola help | grep -q testiso; then
+        cosa kola testiso -S --output-dir ${ARTIFACT_DIR:-/tmp}/kola-testiso --denylist-test iso-offline-install-iscsi* --denylist-test pxe-offline-install.rootfs-appended.bios
+    else
+        # Rerun when failed, use 'unused' tag because of following issue:
+        # https://github.com/coreos/coreos-assembler/issues/4546
+        cosa kola run --output-dir ${ARTIFACT_DIR:-/tmp}/kola-testiso iso.* --rerun --allow-rerun-success tags=unused
+    fi
 }
 
 # Basic syntaxt validation for manifests

--- a/ci/prow-entrypoint.sh
+++ b/ci/prow-entrypoint.sh
@@ -86,17 +86,12 @@ cosa_build() {
 
 # Run all kola tests
 kola() {
+    # Until RHCOS openshift cluster gets updated, iso.* tests run sequentially, so skip them here
     cosa kola run --parallel 2 --output-dir ${ARTIFACT_DIR:-/tmp}/kola --rerun --allow-rerun-success tags=needs-internet --denylist-test iso.*
 
-    # Use 'testiso' for older cosa builds, otherwise use kola.
-    # Until RHCOS openshift cluster gets updated, iso.* tests run sequentially.
-    if cosa kola help | grep -q testiso; then
-        cosa kola testiso -S --output-dir ${ARTIFACT_DIR:-/tmp}/kola-testiso --denylist-test iso-offline-install-iscsi* --denylist-test pxe-offline-install.rootfs-appended.bios
-    else
-        # Rerun when failed, use 'unused' tag because of following issue:
-        # https://github.com/coreos/coreos-assembler/issues/4546
-        cosa kola run --output-dir ${ARTIFACT_DIR:-/tmp}/kola-testiso iso.* --rerun --allow-rerun-success tags=unused
-    fi
+    # Rerun when failed, use 'unused' tag because of following issue:
+    # https://github.com/coreos/coreos-assembler/issues/4546
+    cosa kola run --output-dir ${ARTIFACT_DIR:-/tmp}/kola-testiso iso.* --rerun --allow-rerun-success tags=unused
 }
 
 # Helper function to run the standard build and test workflow

